test/components/views/rooms/SearchBar-test.tsx

```diff
@@ -40,95 +35,65 @@ jest.mock("../../../../src/KeyBindingsManager", () => ({
         searchProps.onSearch.mockClear();
     });
 
-    it("must render child components and pass necessary props", () => {
-        const postHogScreenTracker = wrapper.find(PosthogScreenTracker);
-        const searchWarning = wrapper.find(SearchWarning);
-
-        expect(postHogScreenTracker.length).toBe(1);
-        expect(searchWarning.length).toBe(1);
-        expect(postHogScreenTracker.props().screenName).toEqual("RoomSearch");
-        expect(searchWarning.props().isRoomEncrypted).toEqual(searchProps.isRoomEncrypted);
-        expect(searchWarning.props().kind).toEqual(mockWarningKind);
-    });
-
     it("must not search when input value is empty", () => {
-        const roomButtons = wrapper.find(".mx_SearchBar_button");
-        const searchButton = wrapper.find(".mx_SearchBar_searchButton");
+        const { container } = render(<SearchBar {...searchProps} />);
+        const roomButtons = container.querySelectorAll(".mx_SearchBar_button");
+        const searchButton = container.querySelectorAll(".mx_SearchBar_searchButton");
 
-        expect(roomButtons.length).toEqual(4);
+        expect(roomButtons.length).toEqual(2);
 
-        searchButton.at(0).simulate("click");
-        roomButtons.at(0).simulate("click");
-        roomButtons.at(2).simulate("click");
+        fireEvent.click(searchButton[0]);
+        fireEvent.click(roomButtons[0]);
+        fireEvent.click(roomButtons[1]);
 
         expect(searchProps.onSearch).not.toHaveBeenCalled();
     });
 
     it("must trigger onSearch when value is not empty", () => {
+        const { container } = render(<SearchBar {...searchProps} />);
         const searchValue = "abcd";
 
-        const roomButtons = wrapper.find(".mx_SearchBar_button");
-        const searchButton = wrapper.find(".mx_SearchBar_searchButton");
-        const input = wrapper.find(".mx_SearchBar_input input");
-        input.instance().value = searchValue;
+        const roomButtons = container.querySelectorAll(".mx_SearchBar_button");
+        const searchButton = container.querySelectorAll(".mx_SearchBar_searchButton");
+        const input = container.querySelector<HTMLInputElement>(".mx_SearchBar_input input");
+        if (input) {
+            input.value = searchValue;
+        }
 
-        expect(roomButtons.length).toEqual(4);
+        expect(roomButtons.length).toEqual(2);
 
-        searchButton.at(0).simulate("click");
+        fireEvent.click(searchButton[0]);
 
         expect(searchProps.onSearch).toHaveBeenCalledTimes(1);
         expect(searchProps.onSearch).toHaveBeenNthCalledWith(1, searchValue, SearchScope.Room);
 
-        roomButtons.at(0).simulate("click");
+        fireEvent.click(roomButtons[0]);
 
         expect(searchProps.onSearch).toHaveBeenCalledTimes(2);
         expect(searchProps.onSearch).toHaveBeenNthCalledWith(2, searchValue, SearchScope.Room);
 
-        roomButtons.at(2).simulate("click");
+        fireEvent.click(roomButtons[1]);
```

Comment:
This seems to click button 0 then 1, it used to click 2?

Reply:
I have literally no idea how it used to work...

<img width="1045" alt="Screenshot 2022-11-18 at 08 46 16" src="https://user-images.githubusercontent.com/769871/202660334-58b391b7-a2ec-4391-b32b-44c54be3008b.png">

The component has two buttons only... my guess is that enzyme returns the last item of the list if you give an out of bound index... but i could not find that info on their documentation.